src/Uno.UI/UI/Xaml/Controls/ToggleSwitch.cs

```diff
@@ -187,13 +188,22 @@ public object Header
 		}
 
 		public static readonly DependencyProperty HeaderProperty =
-			DependencyProperty.Register("Header", typeof(object), typeof(ToggleSwitch), new PropertyMetadata(null)); 
+			DependencyProperty.Register("Header", typeof(object), typeof(ToggleSwitch), new PropertyMetadata(null, (s, e) => ((ToggleSwitch)s)?.OnHeaderChanged(e)));
+
+		private void OnHeaderChanged(DependencyPropertyChangedEventArgs e)
+		{
+			if (_headerContentPresenter != null)
+			{
+				_headerContentPresenter.Visibility = e.NewValue != null ? Visibility.Visible : Visibility.Collapsed;
+			}
+		}
```

Comment:
The change will work if the header is changed after the control template is set (after `OnApplyTemplate`), but if the Header is set before (e.g. set explicitly in XAML) you'll need to update the header from `OnApplyTemplate` as well:

```suggestion
		private void OnHeaderChanged(DependencyPropertyChangedEventArgs e)
		{
			if (_headerContentPresenter != null)
			{
				UpdateHeaderContentVisibility();
			}
		}

		private void UpdateHeaderContentVisibility() => _headerContentPresenter.Visibility = Header != null ? Visibility.Visible : Visibility.Collapsed;

```

Reply:
Yes I discovered this is as ApplyTemplate fires after OnHeaderChanged when running WebAssembly.

Initially I just recreated what already exists for Slider.Header, but clearly more needs to be done. Building and debugging in VS is still a challenge so it's taking awhile.